src/kvstore/kvstore_dist.h

```diff
@@ -181,16 +223,21 @@ class KVStoreDist : public KVStoreLocal {
           RunContext rctx, Engine::CallbackOnComplete cb) {
         // convert to ps keys
         size_t size = recv_buf.shape().Size();
-        PSKV& pskv = EncodeKey(key, size);
+
+        // even if inactive should use compressed_pskv for pull when type is not none
```

Comment:
This doesn't make sense. What does active mean?

Reply:
The issue was this
Active actually means that the initialization of keys is done, and all subsequent pushes will be compressed.

When GC is not NONE, the sharding will be slightly different because originally we do original_size/num_servers, now we do compressed_size/num_servers (when sizes are not multiples of num_servers, rounding will cause issues). So even if GC is currently not active, we should use the second way of sharding, else a server won't have expected shard size.